### ISSUE TITLE
Resumable uploads do not work with Safari nor ios

### ISSUE THREAD

**Issue by jjchiw** [danialfarid/ng-file-upload#1394]
Hi!

I had an error with the flag `resumeChunkSize` trying to upload  from safari and ios devices, the extra fields (`_chunkSize`, `_currentChunkSize`,` _chunkNumber` and `_totalSize`) didn't were added to the `formData`.

So I did a little debugging and read the code and I found this
```
this.isResumeSupported = function () {
    return window.Blob && (window.Blob instanceof Function) && window.Blob.prototype.slice;
  };
```

It seems that in Safari and in ios `window.Blob` is not an instance of `Function` so the flag `resumeSupported` was false.

I changed the code to:

```
this.isResumeSupported = function () {
    return window.Blob && (window.Blob instanceof Object) && window.Blob.prototype.slice;
  };
```

And its working, do you think it's an acceptable change? or do you think it has other consequences that I'm not seeing. I tested in Windows  (Firefox, Chrome Edge) OSX (Safari, ios simulator safari, chrome)

**Comment by danialfarid** [comment]
It should be fine for the purpose of uploading but other Upload.xxx functions may bot work properly specially like `urlToBlob`.

**Comment by danialfarid** [comment]
Fixed at 12.0.3

**Comment by jjchiw** [comment]
Thanks!
